scripts/cvmfsexec-osg-wrapper

```diff
@@ -114,30 +105,39 @@ if [[ -n $CVMFS_HTTP_PROXY ]]; then
     add_or_replace "$cvmfs_local_config" CVMFS_HTTP_PROXY "${CVMFS_HTTP_PROXY}"
 fi
 
-"$workdir"/cvmfsexec/umountrepo -a || :
+# If we have to use this script to mount CVMFS, we can't use images from there anyway.
+export ALLOW_NONCVMFS_IMAGES=true
 
-trap '"$workdir"/cvmfsexec/umountrepo -a; cd "$prevdir"; rm -rf "$workdir"' EXIT
+echo >&2 "*** Running cvmfsexec smoke test:"
+if "$workdir"/cvmfsexec/cvmfsexec -N -- /bin/true; then
+    echo >&2 "*** cvmfsexec smoke test passed: you have the permissions to run cvmfsexec directly"
 
-# Mount repos
-for repo in config-osg.opensciencegrid.org $CVMFSEXEC_REPOS; do
-    "$workdir"/cvmfsexec/mountrepo "$repo" || fail "Unable to mount cvmfs repo $repo"
-done
+    # Note: do not use 'exec'; cleanup won't run
+    export CVMFS_BASE=/cvmfs
+    export GLIDEIN_SINGULARITY_BINDPATH="$CVMFS_BASE:/cvmfs"
+    echo "*** cvmfs available at $CVMFS_BASE" >&2
+    "$workdir"/cvmfsexec/cvmfsexec -N $CVMFSEXEC_REPOS -- "$@"
+else
+    echo >&2 "*** cvmfsexec smoke test failed: will use mountrepo instead"
+    command -v fusermount >& /dev/null || fail "Required command 'fusermount' not found"
 
-export CVMFS_BASE="$workdir"/cvmfsexec/dist/cvmfs
+    "$workdir"/cvmfsexec/umountrepo -a || :
 
-export GLIDEIN_SINGULARITY_BINDPATH="$CVMFS_BASE:/cvmfs"
+    trap '"$workdir"/cvmfsexec/umountrepo -a; cd "$prevdir"; rm -rf "$workdir"' EXIT
 
-# If we have to use this script to mount CVMFS, we can't use images from there anyway.
-export ALLOW_NONCVMFS_IMAGES=true
+    # Mount repos
+    for repo in config-osg.opensciencegrid.org $CVMFSEXEC_REPOS; do
+        if ! "$workdir"/cvmfsexec/mountrepo "$repo"; then
+            echo "$repo mount failed, dumping logs:" >&2
+            tail -n 20 "$workdir"/cvmfsexec/log/"$repo".log >&2
+            fail "Unable to mount cvmfs repo $repo"
+        fi
+    done
 
-echo "*** cvmfs available at $CVMFS_BASE" >&2
+    export CVMFS_BASE="$workdir"/cvmfsexec/dist/cvmfs
+    export GLIDEIN_SINGULARITY_BINDPATH="$CVMFS_BASE:/cvmfs"
+    echo "*** cvmfs available at $CVMFS_BASE" >&2
```

Comment:
Isn't this backwards? This looks like the `cvmfsexec` path and we're in the block where the `cvmfsexec` smoke test failed

Reply:
Nope, look at CVMFS_BASE: it's no longer /cvmfs, but something under $workdir.

Reply:
I was fundamentally confused about the `cvmfsexec` vs `mountrepo` paths, though I think some comments saying that `mountrepo` can't bind mount to `/cvmfs` would've cleared things up a bit for me